plinth/utils.py

```diff
@@ -52,6 +52,11 @@ def non_admin_view(func):
     return func
 
 
-def is_user_admin(user):
+def is_user_admin(request, cached=False):
     """Return whether user is an administrator."""
-    return user.groups.filter(name='admin').exists()
+    if cached and request.session['cache_user_is_admin'] is not None:
```

Comment:
If you are updating your code using `vagrant provision` then you can find the logs in `/var/log/plinth/status.log` after logging into the virtual machine using `vagrant ssh`.

I do a slightly different approach, however.  First I login to the virtual machine using `vagrant ssh; sudo su -`.  Then stop plinth system service using `systemctl stop plinth`.  Then I start plinth on the command line: `cd /vagrant; ./setup.py install; ./run --debug`.  The messages then show up in the terminal.

Here are some tests you can check (generally test each line of code change).
* Anonymous users are not allowed to admin sections
* Admins are allowed to anonymous and admin sections
* Logging out of Admin user removes permissions.
* Logging in to regular user does not give admin section permission
* Logging into admin user gives admin permissions
* Repeatedly calling front page as Anonymous user does not result in SQL queries
* Repeatedly calling front page as Admin user does not result in SQL queries
* Repeatedly calling front page as regular user does not result in SQL query
* Admin capabilities in places where you edited to code is still preserved.
* Admin capabilities are denied to users in all places where you edited to code is still preserved.